src/hotspot/share/opto/phaseX.cpp

```diff
@@ -2269,6 +2269,41 @@ void PhasePeephole::print_statistics() {
 }
 #endif
 
+Node* PhaseLowering::apply_ideal(Node* k, bool can_reshape) {
+  // Run the lowered Ideal method to continue doing transformations on the node, while avoiding existing transforms
```

Comment:
Can you call `lower` here and as a result, you can simply do `lower.optimize()` as the main entry?

Reply:
I would prefer to keep it as-is because `PhaseIterGVN::optimize` does a lot of logic that may not be relevant here (such as IGVN verification and IGV printing). This way we can avoid changes to IGVN in the future accidentally impacting lowering in unexpected ways.

Reply:
I actually think it is a good idea to have verification and printing. Since Lowering does IGVN-like transformations, they should behave in generally the same way. If it turns out that we actually need a separate entry then we can create it then.

Reply:
I see, I can understand the benefit there. I think we'll still need to have a custom entry to collect the nodes to place on the worklist and filter based on platform, but we can use `optimize()` to replace the main loop.